jalib/jassert.cpp

```diff
@@ -46,6 +43,9 @@ int jassert_quiet = 0;
 static int theLogFileFd = -1;
 static int errConsoleFd = -1;
 
+static jalib::string& tmpDir() {static jalib::string s;return s;};
+static jalib::string& uniquePidStr() {static jalib::string s;return s;};
```

Comment:
This whole library uses heavy C++ so I wouldn't worry about the intricacies.  C++ only paradigms aren't bad and I would _strongly_ suggest that we keep it the way it is. After all we are not talking about rewriting it all to use C++11, ;-).

The problem with creating a static variable is that it gets us into weird dependency problems during initialization. This particular invocation (static variable inside a function body) is a well known and accepted strategy to avoid polluting the global namespace and weird dependency chains.